src/BPGates.jl

```diff
@@ -518,6 +534,80 @@ function QuantumClifford.apply!(state::BellState, g::PauliNoiseOp)
     return state
 end
 
+# Type for abstracting away noisy operations on bell states.
+abstract type AbstractNoiseBellOp <: BellOp end
+
+"""
+PauliZOp(idx, pz) causes qubit at idx `idx` to have a Z flip with probabilty `pz`.
+"""
+struct PauliZOp <: AbstractNoiseBellOp
```

Comment:
I would not call this an `Op` as this implies it is a deterministic gate, not a noise process that happens only probabilistically.

Reply:
I see. But isn't "PauliNoiseOp" in the original BPGates.jl also a probabilistic application of an X, Y, or Z gate? So I think it's in agreement with that convention?

Technically this struct is just a special case of the PauliNoiseOp but px = 0, py = 0, and pz = pz, but I wanted to create my own for testing purposes (in case PauliNoiseOp was broken for whatever reason; the original BPGates.jl does not have explicit tests for the PauliNoiseOp struct